src/map/zone.cpp

```diff
@@ -582,7 +582,7 @@ void CZone::updateCharLevelRestriction(CCharEntity* PChar)
     {
         // If the level restriction is already the same then no need to change it
         CStatusEffect* statusEffect = PChar->StatusEffectContainer->GetStatusEffect(EFFECT_LEVEL_RESTRICTION);
-        if (statusEffect->GetPower() == m_levelRestriction)
+        if (statusEffect == nullptr || statusEffect->GetPower() == m_levelRestriction)
```

Comment:
shouldn't be physically possible to get here and have that be null, since it was just used in the conditional above

Reply:
Agreed, but 3330 was also raised in a case where I thought it would be impossible to hit.  This is just adding additional safety per the warnings raised.

Reply:
I think something else has to be going on, `HasStatusEffect` would have returned false, and I can't believe it was somehow be destroyed after passing that check but before the `GetStatusEffect` would execute.